threadPool.cpp

```diff
@@ -0,0 +1,48 @@
+#include "threadPool.h"
+
+
+ThreadPool::ThreadPool():
+	done(false), joiner(threads)
+{
+	unsigned const thread_count =
+		std::thread::hardware_concurrency();
+	try 
+	{
+		for (unsigned i = 0; i < thread_count; i++)
+		{
+			threads.push_back(
+				std::thread(&ThreadPool::workThread, this));
+		}
+	}
+	catch (...)
+	{
+		done = true;
```

Comment:
What is the purpose of dropping this flag to true\false when the constructor raises an exception?